### ISSUE TITLE
Distance calculation optimization in distance.py

### ISSUE THREAD

**Issue by LittleLerry** [aflgo/aflgo#140]
In distance.py file, we need to invoke `dijkstra_path_length()` n*t times to get distance. For example, when is_cg is true, the AFLGO will run:
```
  for n in find_nodes (name):
    d = 0.0
    i = 0
    if is_cg:
      for t in targets:
        try:
          shortest = nx.dijkstra_path_length (G, n, t)
          d += 1.0 / (1.0 + shortest)
          i += 1
        except nx.NetworkXNoPath:
          pass
```
However, If we reverse all edges in graph G, say, G becomes GR. Then to get the distance from one node `n` to some targets `t1,t2,...,tk`, we want to invoke `dijkstra_path_length(G,n,t1),…, dijkstra_path_length(G,n,tk)`. It’s equivalent to invoke `dijkstra_path_length(GR,t1,n),…, dijkstra_path_length(GR,tk,n)`. We want to use this API `nx.dijkstra_predecessor_and _distance(GR, t)` for each target to get the distance list for all other nodes. 
In other words, invoking `nx.dijkstra_predecessor_and _distance(GR, t)` once is is equivalent to invoking `dijkstra_path_length(G,n1,t),...,dijkstra_path_length(G,ns,t)`. By combining `nx.dijkstra_predecessor_and _distance(GR, t1),...,nx.dijkstra_predecessor_and _distance(GR, tk)`, we can reduce the calls from `n*t` times to only `t` times. I hope this idea works.
